File: indy-dev/src/getDid.js
const sdk = require('indy-sdk');
const IndyError = require('indy-sdk/src/IndyError');
// const sys =require('sys');
// const colors = require('./colors')
const utils = require('./utils')
const log = console.log;



const poolName = utils.POOL_NAME;


// const walletName = process.argv[2];
// const walletKey = process.argv[3];
// const walletConfig = {"id": walletName}; 
// const walletCredentials = { "key": walletKey }; 
// 
async function wait() {
  await new Promise(resolve => setTimeout(resolve, 1000));

  return 10;
}

const getNymResponse = async (masterDid, stdDid) => {
  await new Promise(resolve => setTimeout(resolve, 1000));
  const submitterDid = masterDid;
  const targetDid = stdDid;
  await sdk.setProtocolVersion(utils.PROTOCOL_VERSION)

  let genesisFilePath = await utils.getPoolGenesisTxnPath(poolName);

  log('genesis_txn:', genesisFilePath)
  const poolConfig = {'genesis_txn': JSON.stringify(genesisFilePath)};
  let poolHandle;

  try {

    try{
    log('\n1. Open pool ledger and get handle\n')
    poolHandle = await sdk.openPoolLedger(poolName, poolConfig);
    }catch (e) {
        if (e.message !== 'PoolLedgerInvalidPoolHandle') {
          console.warn('create wallet failed with message: ' + e.message);
          throw e;
        }
    }
    log('\n2. Open secure wallet\n')
  
    // try {
    //   var walletHandle = await sdk.openWallet(walletConfig, walletCredentials)
    // } catch (e) {
    //   if (e.message !== 'WalletAlreadyExistsError') {
    //     console.warn('create wallet failed with message: ' + e.message);
    //     throw e;
    //   }
    // }

    // log('\n2. Open wallet and get handle\n')
    // const walletHandle = await sdk.openWallet(walletConfig, walletCredentials)
    
    // log('\n3. List my DID\n')

    // let myDid = await sdk.listMyDidsWithMeta(walletHandle)
    // const dids = JSON.stringify(myDid);
    // log(dids.split('},{'));

    log('\n3. Submit get-Nym requset\n')

    const getNymTransaction = await sdk.buildGetNymRequest(submitterDid, targetDid)
    const getNymResponse = await sdk.submitRequest(poolHandle, getNymTransaction)
    const parse = await sdk.parseGetNymResponse(getNymResponse)
    
    // log(getNymResponse.result.state_proof);
    log(parse);
    return parse
    
  } catch(err) {
    throw err
  } finally {
    await sdk.closePoolLedger(poolHandle)

  }

  
}

// getNymResponse("LKF5HhqCcmLL3thpjRARFg", "UhMJWK2H6pU1vCX2yFFYS")
module.exports = {
  getNymResponse,
  poolName,
  utils
};

File: indy-scripts/in_docker/get_did.py
import asyncio
import json
import pprint
import sys
import re

from indy import pool, ledger, wallet, did
from indy.error import IndyError, ErrorCode

from utils import get_pool_genesis_txn_path, PROTOCOL_VERSION, add_error, print_log, POOL_NAME
from collections import OrderedDict

pool_name = POOL_NAME
genesis_file_path = get_pool_genesis_txn_path(pool_name)

# Params
wallet_name = sys.argv[1]
wallet_key = sys.argv[2]
file_name = str(wallet_name) + '_student_did.json'

wallet_config = json.dumps({"id": wallet_name})
wallet_credentials = json.dumps({"key": wallet_key})


async def get_did():
    try:
        await pool.set_protocol_version(PROTOCOL_VERSION)

        #1.
        print_log('\n1. Creates a new local pool ledger configuration that is used '
                  'later when connecting to ledger.\n')
        pool_config = json.dumps({'genesis_txn': str(genesis_file_path)})

        try:
            await pool.create_pool_ledger_config(config_name=pool_name, config=pool_config)
        except IndyError as ex:
            add_error(file_name)
            if ex.error_code == ErrorCode.PoolLedgerConfigAlreadyExistsError:
                pass
        
        #2.
        print_log('\n2. Open pool ledger and get handle from libindy\n')
        pool_handle = await pool.open_pool_ledger(config_name=pool_name, config=None)
                
        #3.
        print_log('\n3. Open wallet and get handle from libindy\n')
        wallet_handle = await wallet.open_wallet(wallet_config, wallet_credentials)

        #4.
        print_log('\n4. List my DID\n')
        my_did = await did.list_my_dids_with_meta(wallet_handle)
        list_did = my_did.split("},{")

        #5.
        print_log('\n5. Parse DID and Make "student_did.json" File\n')
        for i in list_did:
            print(i + "\n\n")
            if "StudentID" in i:
                i = re.sub("\[|\'|\]","", i)
                if i.count("{") != 1:
                    i = "{" + i
                elif i.count("}") != 1:
                    i = i + "}"
                print_log(i)
                jsonObject = json.loads(i)
                student_DID = jsonObject.get("did")
                print_log("Student DID: " + student_DID)
                # Export json of Student DID info
                with open(file_name,'w',encoding="utf-8") as make_file:
                    jsonObject['verkey'] = "None"
                    jsonObject['error'] = "None"
                    json.dump(jsonObject, make_file, ensure_ascii=False, indent="\t")
                break
            else:
                add_error(file_name)
        
        print_log('\n[End of Process]\n')
        
        
            
    except IndyError as e:
        add_error(file_name)
        print('Error occurred: %s' % e)


def main():
    loop = asyncio.get_event_loop()
    loop.run_until_complete(get_did())
    loop.close()


if __name__ == '__main__':
    main()



